packages/framework-core/src/services/graphql/graphql-query-generator.ts

```diff
@@ -141,75 +131,120 @@ export class GraphQLQueryGenerator {
       })
     )
   }
+  public generateFilterQueriesFields(name: string, type: TargetTypeMetadata): GraphQLFieldConfigArgumentMap {
+    const filterArguments = this.generateFilterArguments(type)
+    const filter: GraphQLInputObjectType = new GraphQLInputObjectType({
+      name: `${name}Filter`,
+      fields: () => ({
+        ...filterArguments,
+        ...{ and: { type: new GraphQLList(filter) } },
+        ...{ or: { type: new GraphQLList(filter) } },
+        ...{ not: { type: filter } },
+      }),
+    })
+    return { filter: { type: filter } }
+  }
 
   public generateFilterArguments(typeMetadata: TargetTypeMetadata): GraphQLFieldConfigArgumentMap {
     const args: GraphQLFieldConfigArgumentMap = {}
     typeMetadata.properties.forEach((prop: PropertyMetadata) => {
-      const graphQLPropType = this.typeInformer.getGraphQLTypeFor(prop.type)
-      if (!this.canFilter(graphQLPropType)) {
-        // TODO: We still don't handle filtering by complex properties
-        return
-      }
+      const primitiveType = this.typeInformer.getPrimitiveExtendedType(prop.typeInfo.type)
+
       args[prop.name] = {
-        type: this.generateFilterFor(prop.type),
+        type: primitiveType === Array ? this.generateArrayFilterFor(prop) : this.generateFilterFor(primitiveType),
       }
     })
     return args
   }
 
-  private canFilter(graphQLType: GraphQLNonInputType): boolean {
-    return graphQLType instanceof GraphQLScalarType && graphQLType != GraphQLJSONObject
-  }
+  private generateArrayFilterFor(property: PropertyMetadata): GraphQLInputObjectType {
+    const filterName = `${property.name}PropertyFilter`
 
-  private generateFilterFor(type: AnyClass): GraphQLInputObjectType {
-    const filterName = `${type.name}PropertyFilter`
     if (!this.generatedFiltersByTypeName[filterName]) {
-      const graphQLValueType = this.typeInformer.getGraphQLTypeFor(type)
+      const propFilters: GraphQLInputFieldConfigMap = {}
+      property.typeInfo.parameters.forEach(() => {
+        propFilters.includes = {
+          ...propFilters.includes,
+          ...{
+            type: GraphQLJSONObject,
+          },
```

Comment:
Why doing this and not instead:
```suggestion
          type: GraphQLJSONObject,
```